apps/artemis-web/pages/api/rmq_action.ts

```diff
@@ -0,0 +1,150 @@
+import authorization from '../../middleware/authorization';
+import nc from 'next-connect';
+import {
+  NextApiRequestExtended,
+  NextApiResponseExtended,
+} from '../../definitions';
+import auth from '../../middleware/auth';
+import * as Amqp from 'amqp-ts';
+import { BrokerExchangeOptions, BrokerQueueOptions } from 'typescript-rabbitmq';
+import Broker from 'typescript-rabbitmq';
+
+const sendRMQAction = async (obj) => {
+  const { exchangeName, payload, routing_key } = obj;
+  const {
+    RABBITMQ_USER,
+    RABBITMQ_PASS,
+    RABBITMQ_HOST,
+    RABBITMQ_PORT,
+  } = process.env;
+
+  let config: any = {
+    connection: {
+      user: RABBITMQ_USER,
+      pass: RABBITMQ_PASS,
+      host: '172.18.0.16',
+      port: RABBITMQ_PORT,
+      timeout: 2000,
+      name: 'rabbitmq',
+    },
+    exchanges: [],
+    queues: [],
+    binding: [],
+    logging: {
+      adapters: {
+        stdOut: {
+          level: 3,
+          bailIfDebug: true,
+        },
+      },
+    },
+  };
+
+  let broker = new Broker(config);
+  await broker.connect();
+
+  await broker.addExchange(exchangeName, 'direct', {
+    durable: false,
```

Comment:
maybe also the delivery mode is missing here? (see old `actions.py` implementation). The default might work, but keep in mind that this is a probable reason if you do not see received messages.